tests/org/mozilla/test/ClassCast.java

```diff
@@ -0,0 +1,27 @@
+package org.mozilla.test;
+
+import gnu.testlet.*;
+
+public class ClassCast implements Testlet {
+    public int getExpectedPass() { return 1; }
+    public int getExpectedFail() { return 0; }
+    public int getExpectedKnownFail() { return 0; }
+
+    private interface Interface {}
+
+    class ClassA implements Interface {}
+
+    abstract class ClassB extends ClassA {}
+
+    class ClassC extends ClassB {}
+
+    static TestHarness th;
+
+    public void test(TestHarness th) {
+        // This test illuminates an issue where ClassB's interfaces are setup
+        // in the runtime before the super class has been loaded.
+        Object o = new ClassC();
+        Interface iface = (Interface)o;
+        th.check(true);
```

Comment:
This is a test for a regression that Brendan saw while developing the patch, so I'm not totally sure what caused it.
I suppose the failure was a ClassCastException (which is the exception that would be thrown under the conditions outlined in the comment), but I can't tell for sure.

I will add a check for ClassCastException anyway.